core/domain/story_services.py

```diff
@@ -241,6 +246,226 @@ def save_new_story(committer_id, story):
+                      story_domain.STORY_NODE_PROPERTY_DESTINATION_NODE_IDS):
+                    story.update_node_destination_node_ids(
+                        change.node_id, change.new_value)
+                elif (change.property_name ==
+                      story_domain.STORY_NODE_PROPERTY_EXPLORATION_ID):
+                    story.update_node_exploration_id(
+                        change.node_id, change.new_value)
+            elif change.cmd == story_domain.CMD_UPDATE_STORY_PROPERTY:
+                if (change.property_name ==
+                        story_domain.STORY_PROPERTY_TITLE):
+                    story.update_title(change.new_value)
+                elif (change.property_name ==
+                      story_domain.STORY_PROPERTY_DESCRIPTION):
+                    story.update_description(change.new_value)
+                elif (change.property_name ==
+                      story_domain.STORY_PROPERTY_NOTES):
+                    story.update_notes(change.new_value)
+                elif (change.property_name ==
+                      story_domain.STORY_PROPERTY_LANGUAGE_CODE):
+                    story.update_language_code(change.new_value)
+            elif (
+                    change.cmd ==
+                    story_domain.CMD_MIGRATE_SCHEMA_TO_LATEST_VERSION):
+                # Loading the story model from the datastore into a
+                # Story domain object automatically converts it to use the
+                # latest schema version. As a result, simply resaving the
+                # story is sufficient to apply the schema migration.
+                continue
+        return story
+
+    except Exception as e:
+        logging.error(
+            '%s %s %s %s' % (
+                e.__class__.__name__, e, story_id, change_list)
+        )
+        raise
+
+
+def _save_story(committer_id, story, commit_message, change_list):
+    """Validates a story and commits it to persistent storage. If
+    successful, increments the version number of the incoming story domain
+    object by 1.
+
+    Args:
+        committer_id: str. ID of the given committer.
+        story: Story. The story domain object to be saved.
+        commit_message: str. The commit message.
+        change_list: list(dict). List of changes applied to a story. Each
+            entry in change_list is a dict that represents a StoryChange.
+
+    Raises:
+        ValidationError: An invalid exploration was referenced in the
+            story.
+        Exception: The story model and the incoming story domain
+            object have different version numbers.
+    """
+    if not change_list:
+        raise Exception(
+            'Unexpected error: received an invalid change list when trying to '
+            'save story %s: %s' % (story.id, change_list))
+
+    # Validate that all explorations referenced by the story exist.
+    exp_ids = []
+    for node in story.story_contents.nodes:
+        exp_ids.append(node.exploration_id)
```

Comment:
Might as well only append if not None, right?

Reply:
Yeah, done.